### DISCUSSION TITLE
Help with hooking into poetry commands

### DISCUSSION THREAD

**Discussion by paljax** [nat-n/poethepoet#190] [Q&A] [answered]
Hi, I want to run some tasks after `poetry install` but only if a specific extras group is installed. So `poetry install -E extra_group` and `poetry install --all-extras` should run the task after installation, but `poetry install` shouldn't run it.

Is that possible? This works for me, but I would like it to run only when I install a specific extra.

```
[tool.poe.poetry_hooks]
post_install = "task-name"
```

Thanks a bunch

**Comment by nat-n** [comment, marked as answer]
hi @drjct,

Making a poetry task hook depend on the arguments to the poetry task is not supported today.

If you think this is important (and you're willing to help) then we could look into enhancing the poetry plugin to enable this. There's probably a way for the plugin to know what arguments we passed to the `poetry install` command (TBC), but then how to configure the hook to depend on this is a bit of a trickier question. Off the top of my head I can't think of an elegant solution. 

An alternative approach would be perhaps to expose the poetry arguments to the hook task, either by forwarding them directly or setting an env var. Would that work for your use case?

I imagine something like:
```toml
[tool.poe.poetry_hooks]
post_install = "task-name ${POETRY_ARGS}"
```

**Reply by paljax** [reply]
Thanks, this gave me a good clue. I ended up doing a conditional pip install based on what was in pip list.
